spec/rubocop/cli/autocorrect_spec.rb

```diff
@@ -1885,4 +1885,52 @@ def do_even_more_stuff
     expect(status).to eq(0)
     expect(source_file.read).to eq(source)
   end
+
+  it 'corrects indentation for a begin/rescue/else/ensure/end block properly' do
+    # interplay between `Layout/IndentationWidth`, `Layout/RescueEnsureAlignment`,
+    # and `Layout/ElseAlignment`.
+    create_file('.rubocop.yml', <<~YAML)
+      Style/FrozenStringLiteralComment:
+        Enabled: false
+      Metrics:
+        Enabled: false
+    YAML
+
+    source_file = Pathname('example.rb')
+    create_file(source_file, <<~RUBY)
+      def my_func
+        puts 'do something outside block'
+        begin
+        puts 'do something error prone'
+        rescue SomeException, SomeOther
+         puts 'wrongly indented error handling'
+        rescue StandardError
+         puts 'another wrongly indented error handling'
+        else
+           puts 'wrongly indented normal case handling'
+        ensure
+            puts 'wrongly indented common handling'
+        end
+      end
+    RUBY
+
+    status = cli.run(['--auto-correct-all'])
```

Comment:
I'm not sure the cause. it seems that the issue does not reproduce even if the product code is changed to the original code.

```diff
% git diff .
diff --git a/lib/rubocop/cop/layout/indentation_width.rb b/lib/rubocop/cop/layout/indentation_width.rb
index 73ff3e482..705c65a4d 100644
--- a/lib/rubocop/cop/layout/indentation_width.rb
+++ b/lib/rubocop/cop/layout/indentation_width.rb
@@ -310,20 +310,12 @@ module RuboCop
         def indentation_to_check?(base_loc, body_node)
           return false if skip_check?(base_loc, body_node)

-          if body_node.rescue_type?
-            check_rescue?(body_node)
-          elsif body_node.ensure_type?
+          if %i[rescue ensure].include?(body_node.type)
             block_body, = *body_node
             return unless block_body
-
-            check_rescue?(block_body) if block_body.rescue_type?
-          else
-            true
           end
-        end

-        def check_rescue?(rescue_node)
-          rescue_node.body
+          true
         end

         def skip_check?(base_loc, body_node)
```

```console
% bundle exec rspec spec/rubocop/cli/autocorrect_spec.rb:1889
(snip)

Finished in 0.19408 seconds (files took 0.85788 seconds to load)
1 example, 0 failures

Randomized with seed 42134
```

Reply:
That test is just there to ensure that autocorrect is still working as expected. The tests in `layout/indentation_width_spec.rb` will fail without the changes.

Reply:
```shell
rubocop test.rb --only Layout/IndentationWidth -a
Inspecting 1 file
C

Offenses:

test.rb:2:1: C: [Corrected] Layout/IndentationWidth: Use 2 (not 0) spaces for indentation.
rescue StandardError


1 file inspected, 1 offense detected, 1 offense corrected
```

Corrects to:
```ruby
def a
  rescue StandardError
  ensure
    a
end
```

Then: 

```shell
rubocop test.rb --only Layout/RescueEnsureAlignment -a
Inspecting 1 file
C

Offenses:

test.rb:2:3: C: [Corrected] Layout/RescueEnsureAlignment: rescue at 2, 2 is not aligned with def a at 1, 0.
  rescue StandardError
  ^^^^^^
test.rb:3:3: C: [Corrected] Layout/RescueEnsureAlignment: ensure at 3, 2 is not aligned with def a at 1, 0.
  ensure
  ^^^^^^

1 file inspected, 2 offenses detected, 2 offenses corrected
```

Corrects to:
```ruby
def a
rescue StandardError
ensure
    a
end
```

Which is back to where we started (except for `a` shifted over, but the loops is for moving `rescue` and `ensure`).

Reply:
Thanks for explanation. That's just my two cents. It looks better if from L1890 to L1897 comments and YAML code are replaced with the following code :-)

```suggestion
    status = cli.run(
      [
        '--auto-correct',
        '--only',
        'Layout/IndentationWidth,Layout/RescueEnsureAlignment,Layout/ElseAlignment'
      ]
    )
```

Reply:
Sounds good!

Reply:
@koic updated, thanks.